tests/reg_tests/test_DVGeometry.py

```diff
@@ -1048,6 +1048,128 @@ def test_spanDV_child(self, train=False):
 
         np.testing.assert_allclose(dIdx["span"], dIdx_FD["span"], atol=1e-15)
 
+    def test_embedding_solver(self):
+        DVGeo = DVGeometry(os.path.join(self.base_path, "../../input_files/fuselage_ffd_severe.xyz"))
+
+        # none of these fail yet...
+        test_points = [
+            # Points that work with the linesearch fix on the pyspline projection code.
+            # These points work with a reasonable iteration count (we have 50 for now).
+            [0.49886, 0.31924, 0.037167],
+            [0.49845, 0.32658, 0.039511],
+            [0.76509, 0.29709, 0.037575],
+            # The list of points below are much more problematic. The new fix can handle
+            # some of them but they require a very large iteration count. The FFD here
+            # is ridiculously difficult to embed, but leaving these points here because
+            # they are a great test of robustness if someone wants to improve the solver.
+            # even more down the line.
+            # [0.76474, 0.30461, 0.039028],
+            # [0.49988, 0.29506, 0.031219],
+            # [0.49943, 0.30642, 0.03374],
+            # [0.49792, 0.33461, 0.042548],
+            # [0.49466, 0.35848, 0.06916],
+            # [0.49419, 0.34003, 0.092855],
+            # [0.49432, 0.33345, 0.09765],
+            # [0.49461, 0.31777, 0.10775],
+            # [0.49465, 0.31347, 0.11029],
+            # [0.62736, 0.31001, 0.037233],
+            # [0.76401, 0.32044, 0.042354],
+            # [0.49322, 0.25633, 0.13751],
+            # [0.49358, 0.26432, 0.13435],
+        ]
+
+        DVGeo.addPointSet(test_points, "test", nIter=50)
+
+        # we evaluate the points. if the embedding fails, the points will not be identical
+        new_points = DVGeo.update("test")
+
+        np.testing.assert_allclose(test_points, new_points, atol=1e-15)
+
+    def test_coord_xfer(self):
```

Comment:
The test looks very solid, the only thing that I would add for sake of completeness is an additional global design variable? There are a few ready to use in `commonUtils`

Reply:
I dont think its necessary. the coordinate transformation does its job even with this single point, and I dont want to spend more time on this. I checked the test fails when you screw up the coord_xfer function or any related machinery in dvgeo.